source/simulator/solver.cc

```diff
@@ -458,7 +483,15 @@ namespace aspect
             SolverControl solver_control(5000, 1e-6 * src.l2_norm(), false, true);
             SolverCG<TrilinosWrappers::MPI::Vector> solver(solver_control);
             //Solve with Schur Complement approximation
-            solver.solve(mp_matrix,
+            auto Op_A=LinearOperator<TrilinosWrappers::MPI::Vector>(system_matrix.block(0,0));
+            auto Op_BT = LinearOperator<TrilinosWrappers::MPI::Vector>(system_matrix.block(0,1));
+            auto Op_B = LinearOperator<TrilinosWrappers::MPI::Vector>(system_matrix.block(1,0));
+            auto Op_C = diag_operator(Op_A,inverse_lumped_mass_matrix);
+            auto matrix = Op_B*Op_C*Op_BT;
```

Comment:
Well, it *isn't* a matrix. It's actually an operator. I think it would be nice to give it a descriptive name, say
```suggestion
            auto Op_A=LinearOperator<TrilinosWrappers::MPI::Vector>(system_matrix.block(0,0));
            auto Op_BT = LinearOperator<TrilinosWrappers::MPI::Vector>(system_matrix.block(0,1));
            auto Op_B = LinearOperator<TrilinosWrappers::MPI::Vector>(system_matrix.block(1,0));
            auto Op_C = diag_operator(Op_A,inverse_lumped_mass_matrix);
            auto B_C_BT = Op_B*Op_C*Op_BT;
```